### PR TITLE
Breadth First Search in Csharp  #2171

### LINKED ISSUE #2171: Breadth First Search
- [x] C
- [x] C++
- [x] C# - @prashansag62
- [ ] CoffeeScript
- [x] Dart - @Rukmini-Meda
- [ ] Go - @Aju100
- [x] Java
- [x] JS
- [ ] Kotlin
- [ ] Php - @srimani-programmer
- [x] Python
- [x] Readme
- [x] Ruby - @HardevKhandhar
- [ ] Rust
- [ ] TypeScript

### PR DESCRIPTION
**Pull request by prashansag62**
Fixes #2171 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- Breadth-First Search Code implemented using queue in C# added

#### Languages Used:
-  C#

#### Files Added:
- bfs_in_Csharp.cs file added

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

File: Breadth_First_Search/BFS_in_Csharp.cs [hunk 1]
```diff
@@ -0,0 +1,128 @@
+using System;
```
Comment (singh-shreya6) [inline Breadth_First_Search/BFS_in_Csharp.cs]: Please add a brief problem description at start of file as multi line comment.

File: Breadth_First_Search/BFS_in_Csharp.cs [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+using System.Collections.Generic;                        // for generic data structures  
+
+namespace GraphBfs
+{
+    class graph
+    {
+       private int vertices;                           // no of vertices in a graph
```
Comment (singh-shreya6) [inline Breadth_First_Search/BFS_in_Csharp.cs]: 4 space indentation at start of each block.

File: Breadth_First_Search/BFS_in_Csharp.cs [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+       private List<Int32>[] adj;                      //declaraing list
+       public graph(int v)                    
+       {
+           vertices=v;
+           adj=new List<Int32>[v];                     //initialising Adjacency List with number of vertices
+           for(int i=0;i<v;i++)
```
Comment (singh-shreya6) [inline Breadth_First_Search/BFS_in_Csharp.cs]: Space around operators, and after semi colon.

File: Breadth_First_Search/BFS_in_Csharp.cs [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+
+namespace GraphBfs
+{
+    class graph
+    {
+       private int vertices;                           // no of vertices in a graph
+       private List<Int32>[] adj;                      //declaraing list
+       public graph(int v)                    
+       {
+           vertices=v;
```
Comment (singh-shreya6) [inline Breadth_First_Search/BFS_in_Csharp.cs]: Space around operator.

File: Breadth_First_Search/BFS_in_Csharp.cs [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+            int v = Convert.ToInt32(Console.ReadLine());    
+
+            graph g=new graph(v);  
+
+            Console.WriteLine("Enter number of edge in a graph");        //Taking input number of edges
+            int e = Convert.ToInt32(Console.ReadLine());  
+
+            for(int i=0;i<e;i++)                                         
+            {   
+                Console.WriteLine("Enter nodes of " + (i+1) +" edge");     //taking input of edges
```
Comment (singh-shreya6) [inline Breadth_First_Search/BFS_in_Csharp.cs]: Please change the place of comment, add it before the code, not on the sides across the file. Also please add spaces around operators across the file.

File: Breadth_First_Search/BFS_in_Csharp.cs [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+4 7
+Enter nodes of 10 edge
+7 8
+Enter nodes of 11 edge
+7 9
+Enter nodes of 12 edge
+8 9
+Breadth First Search of Graph :-
+0 1 2 3 4 5 6 7 8 9
+
```
Comment (singh-shreya6) [inline Breadth_First_Search/BFS_in_Csharp.cs]: One blank line is enough

File: Breadth_First_Search/BFS_in_Csharp.cs [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+2 -> 0,4,5
+3 -> 1,6
+4 -> 2,6,5,7,
+5 -> 2,4
+6 -> 3,4
+7 -> 4,8,9
+8 -> 7,9
+9 -> 7,8 
+
+*/
```
Comment (singh-shreya6) [inline Breadth_First_Search/BFS_in_Csharp.cs]: Please add one blank line at end of file

File: Breadth_First_Search/BFS_in_Csharp.cs [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+                g.addedge(e1,e2);                                          //adding edge
+            }   
+
+            Console.WriteLine("Breadth First Search of Graph :-");
+            g.BFS(0);                                                       //calling BFS function with starting node provided.
+
+        }
+    }
+}
+
```
Comment (singh-shreya6) [inline Breadth_First_Search/BFS_in_Csharp.cs]: One blank line

File: Breadth_First_Search/BFS_in_Csharp.cs (hunk 1 above)
Comment (prashansag62) [reply Breadth_First_Search/BFS_in_Csharp.cs]: @singh-shreya6 Description added

File: Breadth_First_Search/BFS_in_Csharp.cs (hunk 2 above)
Comment (prashansag62) [reply Breadth_First_Search/BFS_in_Csharp.cs]: @singh-shreya6 Indentation corrected

File: Breadth_First_Search/BFS_in_Csharp.cs (hunk 3 above)
Comment (prashansag62) [reply Breadth_First_Search/BFS_in_Csharp.cs]: @singh-shreya6 space around operators added

File: Breadth_First_Search/BFS_in_Csharp.cs (hunk 5 above)
Comment (prashansag62) [reply Breadth_First_Search/BFS_in_Csharp.cs]: @singh-shreya6 Position of comments changed

File: Breadth_First_Search/BFS_in_Csharp.cs (hunk 6 above)
Comment (prashansag62) [reply Breadth_First_Search/BFS_in_Csharp.cs]: @singh-shreya6 Extra blank lines removed

File: Breadth_First_Search/BFS_in_Csharp.cs [hunk 9]
```diff
@@ -0,0 +1,137 @@
...
+        private List<Int32>[] adj;                      
+        public graph(int v)                    
+        {   
+            //initialising Adjacency List with number of vertices
+            vertices=v;
+            adj=new List<Int32>[v];                     
+            for(int i = 0; i < v; i++)
+            {
+                adj[i] = new List<int>();                 
+
```
Comment (singh-shreya6) [inline Breadth_First_Search/BFS_in_Csharp.cs:22]: Please remove extra blank line.

File: Breadth_First_Search/BFS_in_Csharp.cs [hunk 10]
```diff
@@ -0,0 +1,137 @@
...
+        {   
+            bool[] visited = new bool[vertices];          
+            Queue<Int32> queue = new Queue<int>();       
+            visited[v] = true;     
+            // enqueuing first node in queue                       
+            queue.Enqueue(v);                           
+            while(queue.Count!= 0)
+            {
+                // dequeue node from beginning of queue
+                v=queue.Dequeue();
```
Comment (singh-shreya6) [inline Breadth_First_Search/BFS_in_Csharp.cs:43]: Please add space around =

File: Breadth_First_Search/BFS_in_Csharp.cs [hunk 11]
```diff
@@ -0,0 +1,137 @@
...
+
+        // function for Breadth First Search ,starting node as parameter
+        public void BFS(int v)                          
+        {   
+            bool[] visited = new bool[vertices];          
+            Queue<Int32> queue = new Queue<int>();       
+            visited[v] = true;     
+            // enqueuing first node in queue                       
+            queue.Enqueue(v);                           
+            while(queue.Count!= 0)
```
Comment (singh-shreya6) [inline Breadth_First_Search/BFS_in_Csharp.cs:40]: Space around !=

File: Breadth_First_Search/BFS_in_Csharp.cs [hunk 12]
```diff
@@ -0,0 +1,137 @@
...
+
+            for(int i = 0; i < e; i++)                                         
+            {   
+                Console.WriteLine("Enter nodes of " + (i+1) +" edge");     
+                string s = Console.ReadLine();
+
+                int e1 =Convert.ToInt32(s.Split(" ")[0]);                  
+                int e2 = Convert.ToInt32(s.Split(" ")[1]);                 
+
+                g.addedge(e1,e2);
```
Comment (singh-shreya6) [inline Breadth_First_Search/BFS_in_Csharp.cs:80]: Space after ,

File: Breadth_First_Search/BFS_in_Csharp.cs [hunk 13]
```diff
@@ -0,0 +1,137 @@
...
+    class graph
+    {   //no of vertices in a graph
+        private int vertices;   
+        //declaraing list
+        private List<Int32>[] adj;                      
+        public graph(int v)                    
+        {   
+            //initialising Adjacency List with number of vertices
+            vertices=v;
+            adj=new List<Int32>[v];
```
Comment (singh-shreya6) [inline Breadth_First_Search/BFS_in_Csharp.cs:18]: Space around =

File: Breadth_First_Search/BFS_in_Csharp.cs [hunk 14]
```diff
@@ -0,0 +1,137 @@
...
+{
+    class graph
+    {   //no of vertices in a graph
+        private int vertices;   
+        //declaraing list
+        private List<Int32>[] adj;                      
+        public graph(int v)                    
+        {   
+            //initialising Adjacency List with number of vertices
+            vertices=v;
```
Comment (singh-shreya6) [inline Breadth_First_Search/BFS_in_Csharp.cs:17]: space around =

**Comment by singh-shreya6** [comment]
No change from so many days. CLosing this PR> If you want to work on this again, please make a fresh PR @prashansag62

File: Breadth_First_Search/BFS_in_Csharp.cs (hunk 9 above)
Comment (prashansag62) [reply Breadth_First_Search/BFS_in_Csharp.cs]: done

File: Breadth_First_Search/BFS_in_Csharp.cs (hunk 10 above)
Comment (prashansag62) [reply Breadth_First_Search/BFS_in_Csharp.cs]: done

File: Breadth_First_Search/BFS_in_Csharp.cs (hunk 11 above)
Comment (prashansag62) [reply Breadth_First_Search/BFS_in_Csharp.cs]: done

File: Breadth_First_Search/BFS_in_Csharp.cs (hunk 12 above)
Comment (prashansag62) [reply Breadth_First_Search/BFS_in_Csharp.cs]: done

File: Breadth_First_Search/BFS_in_Csharp.cs (hunk 13 above)
Comment (prashansag62) [reply Breadth_First_Search/BFS_in_Csharp.cs]: done

File: Breadth_First_Search/BFS_in_Csharp.cs (hunk 14 above)
Comment (prashansag62) [reply Breadth_First_Search/BFS_in_Csharp.cs]: @singh-shreya6  All changes have been done. Kindly look into it. Thanks

File: Breadth_First_Search/BFS_in_Csharp.cs (hunk 7 above)
Comment (singh-shreya6) [reply Breadth_First_Search/BFS_in_Csharp.cs:133]: Please add one line after */ as it shows as error in git.

**Comment by prashansag62** [comment]
@singh-shreya6 Done . Kindly verify.Thanks

**Comment by singh-shreya6** [comment]
Please squash your commits @prashansag62

**Comment by prashansag62** [comment]
@singh-shreya6, Hey I have squashed my commits. Kindly check once. Thank you.

**Comment by singh-shreya6** [comment]
As you can see, in the second tab of commits here, it still shows commit count as 6, so it has not been squashed properly. Please ensure that commit count is 1 there after squashing.

**Comment by prashansag62** [comment]
@singh-shreya6 ,highly apologetic ,some error occurred earlier. Now I have squashed and its showing 1 commit only.
Kindly check.
Thanks

**Pull request merged**